frontend/luadata.lua

```diff
@@ -0,0 +1,153 @@
+--[[--
+Handles append-mostly data such as KOReader's bookmarks and dictionary search history.
+]]
+
+local LuaSettings = require("luasettings")
+local dump = require("dump")
+local logger = require("logger")
+local util = require("util")
+
+local LuaData = LuaSettings:new{
+    name = "",
+    max_backups = 9,
+}
+--- Opens a LuaData file.
+function LuaData:open(file_path, name)
+    if name then self.name = name end
+    local new = {file=file_path, data={}}
```

Comment:
Does that `name`, stored as `self.name` and not in `new` (the _instance_), prevent having multiple LuaData instances?

Reply:
I'm not quite sure what you mean?

Reply:
```lua
lookup_history = LuaData:open(self.ui.doc_settings.sidecar .. "/lookup_history.lua", "LookupHistory")
-- LuaData.name = "LookupHistory"
bookmarks= LuaData:open(self.ui.doc_settings.sidecar .. "/bookmarks.lua", "Bookmarks")
-- LuaData.name = "Bookmarks"
lookup_history:saveSetting("toto", "tutu")
lookup_history:append() -- or :flush()
```
Wouldn't these last ones use LuaData.name = "Bookmarks" in their `f_out:write(self.name.."Entry")` ?

(Haven't tested, just asking, cause I don't know enough about the magical power of metatable and co:)

Reply:
If you do that you will indeed run into trouble because you're reusing the same instance. You'd have the same problem in LuaSettings.

The correct way to do that would be something like this:

```lua
local lookup_history_luadata = LuaData:new{name="bla"} -- our own separate LuaData instance
local lookup_history = lookup_history_luadata:open(self.ui.doc_settings.sidecar .. "/lookup_history.lua")
```

Perhaps I should remove the `name` argument to `open` to encourage proper instances? It's probably more of a leftover from my earliest tests than a well thought out feature anyway.

Reply:
Of course I could write it so that `:open` does what `:new` does but that doesn't seem like a good idea with regard to internal consistency with other parts of the program (and Lua practices in general).

Reply:
Ah, I got the feeling the setmetatable stuff was to make the `new` table a full instance of LuaData, and so that `open()` was indeed returning a LuaData instance (just like a `:new()` would).
I thought it would just need `name` to be part or `new` for it to be fully standalone.

No favorite way of doing it. It's just that, contrary to our single LuaSetting (G_reader_settings), you will probably have more LuaData, so better be ready for concurrent uses.